### PR TITLE
Fixes #11317: Add a dsc version of file_template technique

### PR DESCRIPTION
**Pull request by Fdall**
https://www.rudder-project.org/redmine/issues/11317

### CODE REVIEW

**Comment by Fdall** [comment]
Need to define a folder to store the templates on windows, do not know how we can manage this folder access

**Comment by Fdall** [comment]
Commit modified

File: techniques/fileDistribution/fileTemplate/1.0/fileTemplate.ps1.st [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    $destinations = @(
+    &FILE_TEMPLATE_AGENT_DESTINATION_PATH:{destination |
+    "&destination&" };separator=","& )
+
+    $posthooks = @(
+    &FILE_TEMPLATE_TEMPLATE_POST_HOOK_COMMAND:{destination |
+    "&destination&" };separator=","& )
+
+    $HashCheckType = "md5"
```
Comment (ncharles) [inline techniques/fileDistribution/fileTemplate/1.0/fileTemplate.ps1.st]: why md5 ? it's pretty bad

File: techniques/fileDistribution/fileTemplate/1.0/fileTemplate.ps1.st [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        $local_classes = Merge-ClassContext $local_classes $(File-From-Shared-Folder -Source $sources[$i] -Destination $templateDestinations -HashType $HashCheckType -ReportId $trackingkey[$i] -TechniqueName $techniqueName  -componentName $componentName -componentKey $destinations[$i] -auditOnly:$auditOnly -Report:$true)
+      }
+      $componentName = "Expand template"
+      #Expand the template
+      $local_classes = Merge-ClassContext $local_classes $(File-From-Template-Mustache -SourceTemplate $templateDestinations -Destination $destinations[$i] -ReportId $trackingkey[$i] -TechniqueName $techniqueName -componentName $componentName -componentKey $destinations[$i] -auditOnly:$auditOnly -Report:$true)
+
+      #Posthook execution
+      $class_prefix = "file_from_template_${destinationsCanon}_repaired"
+      $state = try {
+        if ($posthooks[$i] -like "") {
```
Comment (ncharles) [inline techniques/fileDistribution/fileTemplate/1.0/fileTemplate.ps1.st]: what is the benefit of -like compared to -eq ?

File: techniques/fileDistribution/fileTemplate/1.0/fileTemplate.ps1.st [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+          $PostHookString = "No posthook defined"
+          [ComplianceStatus]::result_na
+        } else {
+          if ($local_classes["classes"].contains($class_prefix)) {
+              $command = $(Invoke-Expression "$posthooks[$i] 2>null")
```
Comment (ncharles) [inline techniques/fileDistribution/fileTemplate/1.0/fileTemplate.ps1.st]: shouldn't you use CommandExecution ?
if not, you need to take audit into account

**Comment by Fdall** [comment]
Commit modified

**Comment by ncharles** [comment]
i don't see what you changed on the PR

File: techniques/fileDistribution/fileTemplate/1.0/fileTemplate.ps1.st (hunk 1 above)
Comment (Fdall) [reply techniques/fileDistribution/fileTemplate/1.0/fileTemplate.ps1.st]: Will change it to sha256

File: techniques/fileDistribution/fileTemplate/1.0/fileTemplate.ps1.st (hunk 3 above)
Comment (Fdall) [reply techniques/fileDistribution/fileTemplate/1.0/fileTemplate.ps1.st]: Good point, I will use the GM

File: techniques/fileDistribution/fileTemplate/1.0/fileTemplate.ps1.st (hunk 2 above)
Comment (Fdall) [reply techniques/fileDistribution/fileTemplate/1.0/fileTemplate.ps1.st]: None, it is even slower in this case

**Comment by Fdall** [comment]
Commit modified

**Pull request merged**